### PR TITLE
Storages: remove ColumnFileTiny cache

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### What is changed and how it works?

```commit-message
remove the code of ColumnFileTiny cache which is useless.
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileInMemory.cpp [hunk 1]
```diff
@@ -108,17 +107,10 @@ Block ColumnFileInMemory::readDataForFlush() const
...
-ColumnPtr ColumnFileInMemoryReader::getPKColumn()
-{
-    memory_file.fillColumns(*col_defs, 1, cols_data_cache);
-    return cols_data_cache[0];
-}
-
-ColumnPtr ColumnFileInMemoryReader::getVersionColumn()
+std::pair<ColumnPtr, ColumnPtr> ColumnFileInMemoryReader::getPKAndVersionColumns()
 {
     memory_file.fillColumns(*col_defs, 2, cols_data_cache);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileInMemory.cpp:112]: Should we turn it to be like this?
```
memory_file.fillColumns(*col_defs, 1, cols_data_cache);
memory_file.fillColumns(*col_defs, 2, cols_data_cache);
```

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileInMemory.cpp:112]: no,

```cpp
void ColumnFileInMemory::fillColumns(const ColumnDefines & col_defs, size_t col_count, Columns & result) const
```

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTinyReader.cpp [hunk 2]
```diff
@@ -37,22 +39,88 @@ std::pair<size_t, size_t> ColumnFileTinyReader::readRows(
...
+    {
+        // decrypt the page data in place
+        size_t data_size = page.data.size();
+        char * data = page.mem_holder.get();
+        tiny_file.file_provider->decryptPage(tiny_file.keyspace_id, data, data_size, tiny_file.data_page_id);
+    }
+
+    for (size_t index = 0; index < num_columns_read; ++index)
+    {
+        // the column is fill with default values.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTinyReader.cpp:97]:
```suggestion
        // the column is filled with default values.
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**